src/components/WalletModal/index.tsx

```diff
@@ -126,101 +131,104 @@ export default function WalletModal({
   confirmedTransactions: string[] // hashes of confirmed
   ENSName?: string
 }) {
-  // important that these are destructed from the account-specific web3-react context
-  const { account, connector, activate, error } = useWeb3React()
+  const dispatch = useAppDispatch()
+  const { connector, error, hooks, account } = useWeb3React()
+  const isActiveMap: Record<Wallet, boolean> = {
+    [Wallet.INJECTED]: hooks.useSelectedIsActive(injected),
+    [Wallet.COINBASE_WALLET]: hooks.useSelectedIsActive(coinbaseWallet),
+    [Wallet.WALLET_CONNECT]: hooks.useSelectedIsActive(walletConnect),
+    [Wallet.FORTMATIC]: hooks.useSelectedIsActive(fortmatic),
+  }
 
   const [walletView, setWalletView] = useState(WALLET_VIEWS.ACCOUNT)
   const previousWalletView = usePrevious(walletView)
 
-  const [pendingWallet, setPendingWallet] = useState<AbstractConnector | undefined>()
-
-  const [pendingError, setPendingError] = useState<boolean>()
+  const [pendingConnector, setPendingConnector] = useState<Connector | undefined>()
+  // Need to pass network as a default case because useSelectedError requires a connector
+  const pendingError = hooks.useSelectedError(pendingConnector || network)
 
   const walletModalOpen = useModalOpen(ApplicationModal.WALLET)
   const toggleWalletModal = useWalletModalToggle()
 
+  const previousConnector = usePrevious(connector)
   const previousAccount = usePrevious(account)
```

Comment:
Because the `account` is linked to the `connector`, can you omit `previousAccount` and use `connection` and `previousConnection` instead?